presto-hive/src/test/java/com/facebook/presto/hive/TestCteExecution.java

```diff
@@ -1227,10 +1187,34 @@ public void testNestedCteWithSameName()
                 "t1 as ( select orderkey k from orders where orderkey < 10 ), t3 as ( select t1.k, t2.k from t1 left join t2 on t1.k=t2.k ) select * from t3 ) " +
                 "select * from t3 except select * from t4";
         QueryRunner queryRunner = getQueryRunner();
-        compareResults(queryRunner.execute(getMaterializedSession(),
-                        testQuery),
-                queryRunner.execute(getSession(),
-                        testQuery));
+        verifyResults(queryRunner, testQuery, ImmutableList.of(new CTEInfoValue("t1", 1, false),
+                new CTEInfoValue("t2", 1, false),
+                new CTEInfoValue("t3", 1, false),
+                new CTEInfoValue("t4", 1, false)));
+    }
+
+    private void verifyResults(QueryRunner queryRunner, String query, List<CTEInfoValue> expectedCTEInfoValues)
+    {
+        Session materializedSession = getMaterializedSession();
+        Session session = getSession();
+        verifyResults(materializedSession, session, queryRunner, query, expectedCTEInfoValues);
+    }
+
+    private void verifyResults(Session materializedSession, Session session, QueryRunner queryRunner, String query, List<CTEInfoValue> expectedCTEInfoValues)
+    {
+        compareResults(queryRunner.execute(materializedSession,
```

Comment:
Some tests use compareResuts with verify ordering set to true, so you might need that info

Reply:
I noticed there isn't any place which calls the compareResults with the checkOrdering field as true yet, which is why I didn't add the boolean checkOrdering field to verifyResults per YAGNI principles. If there are some tests that should be verifying the order, I didn't see them, but I'm okay adding the field for future proofing of the verifyResults method if that is what you want.
<img width="1371" alt="Screenshot 2024-07-23 at 8 09 57 AM" src="https://github.com/user-attachments/assets/a34f37a8-39fd-4881-a1c0-dd9f767ffd5b">

Reply:
Hmm thats strange, There are a couple of tests which use it https://github.com/prestodb/presto/blob/master/presto-hive/src/test/java/com/facebook/presto/hive/TestCteExecution.java#L677

Reply:
Ohh just this one^

Reply:
I see it now. Must have missed one. I will make the changes then.